### PR TITLE
Update search term when navigating with browser`s back and forward buttons

### LINKED ISSUE #2082: Add the search term as a query parameter to the single result page
## Problem

Currently, the single result page re-uses the search term value saved in the store to set the value of the search bar in the header. This works when browsing without the browser navigation. However, if you hit the browser `back` and `forward` buttons, there is no way in the app to understand the navigation direction and set the search term accordingly.

## Description

- [ ] Set the search term as a query parameter of the single result page. 
So, when navigating from the search result for "cat" to the first result, your route would be `/image/f9384235-b72e-4f1e-9b05-e1b116262a29?q=cat`
- [ ] Update the search store search term in `beforeRouteEnter`
Only update if navigating from the "search" page, and the "q" query parameter is available.

## Alternatives

We could also wait until we update to Nuxt 3, which makes it possible to understand whether the user pressed the "forward" or "back" button. However, we would still have to save the search term history somewhere to update it correctly.

## Additional context

For more discussion of this issue, look at the comments in #2046.

### PR DESCRIPTION
**Pull request by anton202**
## Fixes

Fixes #2082  by @obulat 

## Description

implemented the steps that described in the issue:
* add query param with the searcn term to the url of the single result page
* update search term that is saved in the store when navigating from search page 

now, when doing a search from a single result page and then going back with the browser button 
the search term in the header will be set correctly.

<b>The curent State of staging</b>

https://user-images.githubusercontent.com/29175506/212668428-c1ed8f6f-734a-4678-be2a-206e06d89ade.mp4

<b>This pull request</b>

https://user-images.githubusercontent.com/29175506/212669005-ed3fbb7e-fc34-4197-9032-23e3212a750e.mp4

## Testing Instructions
Follow the steps in the video :)
also, try this steps with an audio result

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — COMMENTED** [review]
This looks really good, @anton202! Do you think you could add an e2e test?

**Comment by anton202** [comment]
@obulat sure

**Comment by anton202** [comment]
i dont know why the e2e/homepage.spec.ts test is failing... i didnt changed anything related with that test. but anyway, i`ve changed the css selector of the search type pop over, in that test and now it dose pass.

regarding the e2e/translation-banner.spec.ts 
when i run all the tests, this test fail
when i run only this test, it pass

**Review by obulat — CHANGES_REQUESTED** [review]
It works beautifully, @anton202! And the tests are 💯 

One request I have is to change the `searchTerm` to be a prop for `VImageCellSquare`/`VAudioTrack`/ `VImageCell`, and actually get it from the `searchStore` in the parent components. This way, the `searchStore.searchTerm` will be called once for each of the media items, and the `Cell`/`Track` components will not need to import the `useSearchStore`.

**Comment by anton202** [comment]
sure, you rigth :)

**Review by obulat — APPROVED** [review]
Awesome work, thank you, @anton202! Thank you for adding the changes and the test for them!
I've rebased your branch onto main and force-pushed it, so now the CI is green.

**Comment by anton202** [comment]
@obulat Thank you :)

File: test/playwright/e2e/search-navigation.spec.ts [hunk 1]
```diff
@@ -125,3 +126,97 @@ test.describe("search history navigation", () => {
...
+    })
+
+    /**
+     * # search for cat
+     * # navigate to the first image type result
+     * # search for dog
+     * # go back to single result page of cat
+     * # search term inside the header should be set to cat
+     */
+    test("search term should be set to the search query param on an image type single result page", async ({
```
[the remark this answers is not in the document]
Comment (anton202) [reply test/playwright/e2e/search-navigation.spec.ts:174]: so, does this test pass now?

**Review by zackkrida — APPROVED** [review]
This appears to be working perfectly. Great work @anton202

**Pull request merged**